src/lib.rs

```diff
@@ -594,3 +604,120 @@ where
         writer.write_str("\n")
     }
 }
+
+/// Enforce a singleton to ensure we are the only `FileWatcher` running and
+/// submitting telemetry to InfluxDB
+fn enforce_singleton(filename: &Path) -> Result<Flock<File>> {
+    let lockfile = OpenOptions::new()
+        .create(true)
+        .append(true)
+        .open(filename)?;
+
+    let lock = match Flock::lock(lockfile.try_clone()?, FlockArg::LockExclusiveNonblock) {
+        Ok(lock) => lock,
+        Err((_, Errno::EWOULDBLOCK)) => {
+            // Silently exit as another running `FileWatcher` was found
+            exit(0);
+        }
+        Err((_, e)) => return Err(TelemetryError::from(e)),
+    };
+
+    Ok(lock)
+}
+
+/// Daemonise the current process
+///
+/// This function forks and has the parent immediately return. The forked off
+/// child then follows the common "double-fork" method of daemonising.
+fn daemonise(log_filename: &PathBuf) -> Result<bool> {
+    stdout().flush()?;
+    stderr().flush()?;
+
+    // Return if we are the parent
+    if unsafe { fork()? }.is_parent() {
```

Comment:
do we really need to use unsafe?

Reply:
Yes, unsafe is needed to `fork`, and we need to fork as the Watchers need to live outside of the process running. Otherwise, is the original process exits, tracing won't be recorded to disk.

I think it's fine here however because `unsafe` is being scoped to a single function each time.